### PR TITLE
Disable profiling as the default option when using MSVC

### PR DESCRIPTION
**Pull request by masterleinad**
Partly following up on #2977. In particular, I also need da39c79 to compile with `clang-cl`.

### CODE REVIEW

**Comment by masterleinad** [comment]
Retest this please.

File: tpls/gtest/gtest/gtest.h [hunk 1]
```diff
@@ -48,6 +48,8 @@
 // registration from Barthelemy Dagenais' (barthelemy@prologique.com)
 // easyUnit framework.
 
+#define GTEST_HAS_TR1_TUPLE 0
```
Comment (dalg24) [inline tpls/gtest/gtest/gtest.h:51]: Did you consider conditionally adding the definition from CMake?

Comment (masterleinad) [reply tpls/gtest/gtest/gtest.h:51]: Yes, I'm indifferent about that.

File: cmake/kokkos_enable_options.cmake [hunk 2]
```diff
@@ -45,7 +45,11 @@ UNSET(_UPPERCASE_CMAKE_BUILD_TYPE)
 KOKKOS_ENABLE_OPTION(LARGE_MEM_TESTS      OFF "Whether to perform extra large memory tests")
 KOKKOS_ENABLE_OPTION(DEBUG_BOUNDS_CHECK   OFF "Whether to use bounds checking - will increase runtime")
 KOKKOS_ENABLE_OPTION(COMPILER_WARNINGS    OFF "Whether to print all compiler warnings")
-KOKKOS_ENABLE_OPTION(PROFILING            ON  "Whether to create bindings for profiling tools")
+IF (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" OR "x${CMAKE_CXX_SIMULATE_ID}" STREQUAL "xMSVC")
+  KOKKOS_ENABLE_OPTION(PROFILING          OFF  "Whether to create bindings for profiling tools")
+ELSE()
+  KOKKOS_ENABLE_OPTION(PROFILING          ON  "Whether to create bindings for profiling tools")
+ENDIF()
```
Comment (dalg24) [inline cmake/kokkos_enable_options.cmake:52]:
```suggestion
IF (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" OR "x${CMAKE_CXX_SIMULATE_ID}" STREQUAL "xMSVC")
  SET(PROFILING_DEFAULT OFF)
ELSE()
  SET(PROFILING_DEFAULT ON)
ENDIF()
KOKKOS_ENABLE_OPTION(PROFILING ${PROFILING_DEFAULT} "Whether to create bindings for profiling tools")
```
for consistency

File: tpls/gtest/gtest/gtest.h (hunk 1 above)
Comment (dalg24) [reply tpls/gtest/gtest/gtest.h:51]: You must 
* either clearly comment that you added that line and state why you did so (the commit message does not contain any useful information)
* or conditionally add a compile definition to the `kokkos_gtest` target

Comment (dalg24) [reply tpls/gtest/gtest/gtest.h:51]: I believe the right thing to do is the compile definition.
We already add `-DGTEST_HAS_PTHREAD=0`
https://github.com/kokkos/kokkos/blob/785d19f23ed4ba43354f9ecd7ef67cd5976ca4c8/core/unit_test/CMakeLists.txt#L17-L22

File: cmake/kokkos_tpls.cmake [hunk 3]
```diff
@@ -13,7 +13,12 @@ KOKKOS_TPL_OPTION(LIBNUMA Off)
 KOKKOS_TPL_OPTION(MEMKIND Off)
 KOKKOS_TPL_OPTION(CUDA    Off)
 KOKKOS_TPL_OPTION(LIBRT   Off)
-KOKKOS_TPL_OPTION(LIBDL   On)
+MESSAGE(STATUS "COMPILER ID ${CMAKE_CXX_COMPILER_ID}")
+IF (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" OR "x${CMAKE_CXX_SIMULATE_ID}" STREQUAL "xMSVC")
```
Comment (crtrott) [inline cmake/kokkos_tpls.cmake:17]: how about `if(WIN32)`? That would also catch the "we use nvcc as the compiler on windows" case

**Review by crtrott — DISMISSED** [review]

File: cmake/kokkos_tpls.cmake [hunk 4]
```diff
@@ -13,7 +13,12 @@ KOKKOS_TPL_OPTION(LIBNUMA Off)
 KOKKOS_TPL_OPTION(MEMKIND Off)
 KOKKOS_TPL_OPTION(CUDA    Off)
 KOKKOS_TPL_OPTION(LIBRT   Off)
-KOKKOS_TPL_OPTION(LIBDL   On)
+MESSAGE(STATUS "COMPILER ID ${CMAKE_CXX_COMPILER_ID}")
```
Comment (dalg24) [inline cmake/kokkos_tpls.cmake:16]:
```suggestion
```

**Review by crtrott — APPROVED** [review]

**Pull request merged**